src/components/molecules/SecretPasswordForm/SecretPasswordForm.jsx

```diff
@@ -20,22 +21,27 @@ const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
     setError(false);
   }
 
-  function passwordSubmitted(e) {
+  const passwordSubmitted = async (e) => {
     e.preventDefault();
     setMessage("Checking password...");
 
-    const checkPassword = firebase.functions().httpsCallable("checkPassword");
-    checkPassword({ venue: venueId, password: password })
-      .then(() => {
-        setInvalidPassword(false);
-        setMessage("Password OK! Proceeding...");
-        history.push(venueEntranceUrl(venueId));
-      })
-      .catch(() => {
-        setInvalidPassword(true);
-        setMessage(null);
+    try {
+      const result = await firebase
+        .functions()
+        .httpsCallable("access-checkAccess")({
+        venueId,
+        password,
       });
-  }
+      localStorage.setItem(getAccessTokenKey(venueId), result.data.token);
```

Comment:
This would probably be better wrapped up as a function in `utils/localStorage` or maybe something related to `auth` or similar?